changelog/next/changes/4916--timestamps-and-durations.md

```diff
@@ -0,0 +1,9 @@
+Timestamps are now printed with a `Z` suffix to indicate that they are relative
+to UTC. Furthermore, the fractional part of the seconds is no longer always
+printed using 6 digits. Instead, timestamps that do not have sub-second
+information no longer have a fractional part. Other timestamps are either
+printed with 3, 6 or 9 fractional digits, depending on their resolution.
```

Comment:
Unfortunately, this part requires adapting the Tenzir Platform:

```
Failed to get pipeline metrics: ParseError

ReadonlyArray<MetricsIngressEgressFromMetricsPipeline>
└─ [0]
   └─ MetricsIngressEgressFromMetricsPipeline
      └─ [\"timestamp\"]
         └─ ((string <-> string) <-> Date)
            └─ Type side transformation failure
               └─ Date
                  └─ Predicate refinement failure
                     └─ Expected Date, actual Invalid Date"
```

I suppose we can easily make the Z optional in the Platform code base, as right now it appears that the Z is always assumed not to be there.

Reply:
Waiting for https://github.com/tenzir/app/pull/866 now.